clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -907,15 +907,32 @@ private void throwIfProducerClosed() {
                     compressionType, serializedKey, serializedValue, headers);
             ensureValidRecordSize(serializedSize);
             long timestamp = record.timestamp() == null ? time.milliseconds() : record.timestamp();
-            log.trace("Sending record {} with callback {} to topic {} partition {}", record, callback, record.topic(), partition);
+            if (log.isTraceEnabled()) {
+                log.trace("Sending record {} with callback {} to topic {} partition {}", record, callback, record.topic(), partition);
```

Comment:
Maybe "attempting to append " would be better than "sending", since we might fail to append.